geode-tcp-server/src/main/java/org/apache/geode/distributed/internal/tcpserver/InetSocketWrapper.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.geode.distributed.internal.tcpserver;
+
+import java.net.InetAddress;
+import java.net.InetSocketAddress;
+import java.util.Objects;
+
+import org.apache.commons.validator.routines.InetAddressValidator;
+
+class InetSocketWrapper {
+  String hostName;
+  protected InetSocketAddress inetSocketAddress;
+  private InetSocketAddress attemptedToResolve;
+
+  protected InetSocketWrapper() {}
+
+  protected InetSocketWrapper(String hostName, int port) {
+    this.hostName = hostName;
+    if (hostName == null) {
+      inetSocketAddress = new InetSocketAddress(port);
+    } else if (InetAddressValidator.getInstance().isValid(hostName)) {
+      // numeric address - use as-is
+      inetSocketAddress = new InetSocketAddress(hostName, port);
+    } else {
+      // non-numeric address - resolve hostname when needed
+      inetSocketAddress = InetSocketAddress.createUnresolved(hostName, port);
+    }
+  }
+
+  public String getHostName() {
+    if (hostName != null) {
+      return hostName;
+    }
```

Comment:
This checking is superfluous because `InetSocketAddress.getHostName()` does this same `null` check.

Reply:
I found that this code is necessary to preserve the original string passed in to the constructor.  If you look at the constructor for InetSocketAddress you'll see that it discards the hostname parameter if it's a numeric address.